crates/contract/tests/sandbox/common.rs

```diff
@@ -725,3 +726,95 @@ pub async fn submit_tee_attestations(
     }
     Ok(())
 }
+
+pub async fn call_contract_key_generation(
+    domains_to_add: Vec<DomainConfig>,
+    accounts: &[Account],
+    contract: &Contract,
+    expected_epoch_id: u64,
+) {
+    for (domain_attempt, domain) in domains_to_add.into_iter().enumerate() {
+        for account in accounts {
+            check_call_success(
+                account
+                    .call(contract.id(), "vote_add_domains")
+                    .args_json(json! ({
+                        "domains": vec![domain.clone()],
+                    }))
+                    .transact()
+                    .await
+                    .unwrap(),
+            );
+        }
+
+        let state: ProtocolContractState = contract.view("state").await.unwrap().json().unwrap();
+        match state {
+            ProtocolContractState::Initializing(state) => {
+                assert_eq!(state.domains.domains().len(), domain_attempt + 1);
+            }
+            _ => panic!("should be in initializing state"),
+        };
+
+        check_call_success(
+            accounts[0]
```

Comment:
For this to succeed, we need `accounts[0]` to be the participant with the lowest `ParticipantId`.
There should be a function comment stating this.